restart_process/Tiltfile

```diff
@@ -62,10 +64,9 @@ def _helper(base_ref, ref, entrypoint, live_update, restart_file=RESTART_FILE, t
 
 def docker_build_with_restart(ref, context, entrypoint, live_update,
                               base_suffix='-tilt_docker_build_with_restart_base', restart_file=RESTART_FILE,
-                              trigger=None, **kwargs):
+                              trigger=None, entr_flags=None, **kwargs):
```

Comment:
Thanks for the PR @mhaley-tignis !

I'm a little bit concerned about whether people will be able to use this API successfully. You really need to deeply understand both how entr works and how your underlying live-reloader works

I remember it being a deep discussion when we wrote this extension. AFAIR, some live-reload entrypoints only exit on catastrophic error (i.e., live-reload isn't possible at all).

@landism do you remember the details of this discussion?

Reply:
`-z` will be there so that the container exits when the process finishes. This is useful for job containers, or init containers that you expect to exit when done.

Reply:
i wonder if it makes sense for the api here to be something like `restart_policy="always"` or `restart_on_exit=True`?

in any case, lemme take care of the container image separately

Reply:
> @landism do you remember the details of this discussion?

sorry, doesn't ring a bell and I don't see any mention of it in the doc that was written at the time

hiding entr flags from the user sounds good, though

Reply:
I like the idea of using `restart_policy="always"` because it is more flexible for future expansion if it is ever needed, but I think that it is confusing from a user perspective. As a user I could get confused and think the `restart_policy` applies to _when_ tilt decides to restart the entrypoint, same with `restart_on_exit`. We really aren't restarting anything, maybe we want something like `proc_exit_policy`?

Reply:
how about `exit_policy="restart"` for the new behavior? and can you add this to the custom_build_with_restart function while you're here?